Gems/AtomTressFX/Code/Rendering/HairRenderObject.cpp

```diff
@@ -0,0 +1,1201 @@
+            //!
+            //!                                    Update Methods
+            //!
+            //!-------------------------------------------------------------------------------------
+            
+            // [TODO] Possibly move wind settings to Simulation Parameters or alike.
+
+            // Wind is in a pyramid around the main wind direction.
+            // To add a random appearance, the shader will sample some direction
+            // within this cone based on the strand index.
+            // This function computes the vector for each edge of the pyramid.
+            // [To Do] Requires more testing
+            static void SetWindCorner(
+                Quaternion rotFromXAxisToWindDir, Vector3 rotAxis,
+                float angleToWideWindCone, float wM, AMD::float4& outVec)
+            {
+                static const Vector3 XAxis(1.0f, 0, 0);
+                Quaternion rot(rotAxis, angleToWideWindCone);
+                // original code: Vector3 newWindDir = rotFromXAxisToWindDir * rot * XAxis;
+                Vector3 newWindDir = (rotFromXAxisToWindDir * rot).TransformVector(XAxis);  // test
+                outVec.x = newWindDir.GetX() * wM;
+                outVec.y = newWindDir.GetY() * wM;
+                outVec.z = newWindDir.GetZ() * wM;
+                outVec.w = 0;  // unused.
+            }
+
+            void HairRenderObject::SetWind(const Vector3& windDir, float windMag, int frame)
+            {
+                float wM = windMag * (pow(sin(frame * 0.01f), 2.0f) + 0.5f);
+
+                Vector3 windDirN(windDir);
+                windDirN.Normalize();
+
+                Vector3 XAxis(1.0f, 0, 0);
+                Vector3 xCrossW = XAxis.Cross(windDirN);
+
+                // Adi: this part is now using AZ::Quaternion - verify correctness
+                Quaternion rotFromXAxisToWindDir;
+                rotFromXAxisToWindDir.CreateIdentity();
+
+                float angle = asin(xCrossW.GetLength());
+
+                if (angle > 0.001)
+                {
+                    // original code: rotFromXAxisToWindDir.SetRotation(xCrossW.Normalize(), angle);
+                    rotFromXAxisToWindDir.CreateFromVector3AndValue(xCrossW.GetNormalized(), angle);
+                }
+
+                const float angleToWideWindCone = AZ::DegToRad(40.f);
+
+                SetWindCorner(rotFromXAxisToWindDir, Vector3(0, 1.0, 0),  angleToWideWindCone, wM, m_simCB->m_Wind);
+                SetWindCorner(rotFromXAxisToWindDir, Vector3(0, -1.0, 0), angleToWideWindCone, wM, m_simCB->m_Wind1);
+                SetWindCorner(rotFromXAxisToWindDir, Vector3(0, 0, 1.0),  angleToWideWindCone, wM, m_simCB->m_Wind2);
+                SetWindCorner(rotFromXAxisToWindDir, Vector3(0, 0, -1.0), angleToWideWindCone, wM, m_simCB->m_Wind3);
+                // fourth component unused. (used to store frame number, but no longer used).
```

Comment:
This was the original TressFX remark and by removing it I reduce the possibility to match to original if ever needed (since so much code shifted and was reconstructed).